compiler/testData/diagnostics/tests/declarationChecks/kClassInSignature.fir.kt

```diff
@@ -1,23 +1,23 @@
 // !DIAGNOSTICS: -TYPE_PARAMETER_AS_REIFIED -TYPE_PARAMETER_OF_PROPERTY_NOT_USED_IN_RECEIVER -UNUSED_VARIABLE -UNUSED_PARAMETER
 
-fun <T> test1() = <!OTHER_ERROR!>T<!>::class
-fun <T : Any> test2() = <!OTHER_ERROR!>T<!>::class
+fun <T> <!KCLASS_WITH_NULLABLE_TYPE_PARAMETER_IN_SIGNATURE!>test1<!>() = T::class
+fun <T : Any> test2() = T::class
 
-val <T> test3 = <!OTHER_ERROR!>T<!>::class
-val <T> test4 get() = <!OTHER_ERROR!>T<!>::class
+val <T> <!KCLASS_WITH_NULLABLE_TYPE_PARAMETER_IN_SIGNATURE!>test3<!> = T::class
+val <T> <!KCLASS_WITH_NULLABLE_TYPE_PARAMETER_IN_SIGNATURE!>test4<!> get() = T::class
 
-fun <T> test5() = listOf(<!OTHER_ERROR!>T<!>::class)
+fun <T> test5() = listOf(T::class)
 
-fun <T> test6(): kotlin.reflect.KClass<T> = <!OTHER_ERROR!>T<!>::class
-fun <T> test7(): kotlin.reflect.KClass<*> = <!OTHER_ERROR!>T<!>::class
+fun <T> test6(): kotlin.reflect.KClass<T> = T::class
```

Comment:
So, this one is the only regression for now. It should be `UPPER_BOUND_VIOLATED`, but I'd like to make a check point here, and will see related diagnostics together.

Reply:
Looks like we already have an expression-level checker: https://github.com/JetBrains/kotlin/blob/master/compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/expression/FirUpperBoundViolatedChecker.kt but the same idea should be applied to the return type of declarations. May need more study/refactoring.